.github/workflows/compare-ds.yaml

```diff
@@ -106,6 +106,26 @@ jobs:
             ${{ steps.diff.outputs.log }}
             ```
 
+            </details>
+          edit-mode: replace
+      - name: Create or update a trimmed comment
+        if: ${{ steps.compare_ds.outputs.COMPARE_DS_OUTPUT_SIZE > 65000 }}
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            This datastream diff is auto generated by the check `Compare DS/Generate Diff`.
+            Due to the excessive size of the diff, it has been trimmed to fit the 4096-character limit.
```

Comment:
The limit is no longer 4096 characters.
It is probably based on 16-bit integer (0 to 65535), so setting the limit body to 65000 is sufficient - the surrounding text can't get longer that easily.
Initially I thought that 8000+ characters is too much, but I don't think so any more, the diff in the example PR is perfectly usable.

Reply:
Fixed by amending the commit. The limit is somehow connected to: https://github.com/actions/virtual-environments/issues/3257 but I haven't really spend too much time investigating it. I believe the 65k chars is enough for our purposes.